aiqleads/core/project_tracking.py

```
@@ -60,8 +124,13 @@ def register_component(self,
                 logger.error(f"Invalid component path: {component_path}")
                 return False
                 
+            # Check for duplicates
+            if self._get_component_id(component_path) in self.components:
+                logger.error(f"Component already registered: {component_path}")
```

Ruff (0.8.2)

Logging statement uses f-string

(G004)